### ISSUE TITLE
ordering arrays

### ISSUE THREAD

**Issue by daimor** [intersystems-ru/cos-guidelines#11]
instead of with getting next index in the end of loop

```
set ind=$order(array(""))
for {
    quit:ind=""

    /// some code here
    set ind=$order(array(ind))
}
```

should be

```
set ind=""
for {
    set ind=$order(array(ind))
    quit:ind=""

    /// some code here
}
```

**Comment by adaptun** [comment]
IMHO, it's more like best practices, not code style guidelines.

**Comment by tsafin** [comment]
Agreed with Sasha @adaptun here - it's more suitable to future "Best Practices" Wiki. Same as with #10

**Comment by isc-solon** [comment]
Both loops above are fine. I wouldn't recommend one more than the other. The first style gives you the ability to quit **before** entering the loop, but there are 2 $order functions. The second style only has one $order function, but you **always** enter the loop.

**Comment by daimor** [comment]
In any way you entering the loop, and in second style, you also can quit before `$order`, but I don't see any way why it should be needed. One main idea, is, in first variant, you can't use `continue`, because you should first get a next value by `$order`, and after that you code will be unreadable. And in "right" variant, you can go to next value in any place of your loop.

```
set ind=""
for {
    set ind=$order(array(ind))
    quit:ind=""
    quit:ind="end"
    continue:$get(array(ind,"skip"))

    /// some code here
}
```

**Comment by eduard93** [comment]
We can start filling `BESTPRACTICES.md` as well

**Comment by tsafin** [comment]
@eduard93 we could start witth MD file, but I still believe that Wiki pages would be easier to maintain and navigate for this kind of content.

**Comment by eduard93** [comment]
@tsafin GitHub project wiki eg. [MDX2JSON wiki](https://github.com/intersystems-ru/Cache-MDX2JSON/wiki) is just another repo with md files, so I see no point in maintaining two repos with text files instead of one.
![default](https://cloud.githubusercontent.com/assets/5127457/11983528/251e9fd0-a9c5-11e5-8416-d8d2a08a4a2a.PNG)

**Comment by daimor** [comment]
Or instead of wiki possible to use [Read the docs](https://readthedocs.org/), one of examples [here](https://wopi.readthedocs.org/en/latest/)

**Comment by tsafin** [comment]
Although their markup might be the same, but from usabiliy point of view Wiki is much more convenient to use (especially for well structured hierarchy). IMVHO

**Comment by tsafin** [comment]
@daimor but Read the docs looks nice. But this is external dependency, which might eventually 
disappear.

P.S.
If time permits we need to play with this tool though.
